pymodbus/client/base.py

```diff
@@ -189,25 +189,27 @@ async def async_execute(self, request=None):
         """Execute requests asynchronously."""
         request.transaction_id = self.transaction.getNextTID()
         packet = self.framer.buildPacket(request)
-        self.transport_send(packet)
-        if self.params.broadcast_enable and not request.slave_id:
-            resp = b"Broadcast write sent - no response expected"
-        else:
-            count = 0
-            while count <= self.params.retries:
-                try:
-                    req = self._build_response(request.transaction_id)
-                    resp = await asyncio.wait_for(
-                        req, timeout=self.comm_params.timeout_connect
-                    )
-                    break
-                except asyncio.exceptions.TimeoutError:
-                    count += 1
-            if count > self.params.retries:
-                self.close(reconnect=True)
-                raise ModbusIOException(
-                    f"ERROR: No response received after {self.params.retries} retries"
+
+        count = 0
```

Comment:
This is not correct, retries is "retry to read" not to send. We cannot automatically send a request multiple time, at least not all requests.

Example: App sends a write request to step up a motor 1 step, if you repeat that it might be fatal. Some devices act on read requests and e.g. initiate an expensive remote request.


Only the app knows if it is safe to resend a request, so we have on purpose left that to the app.